test/util/util.js

```diff
@@ -80,44 +82,50 @@ function mockAjax(pairs) {
     setNextPair(pairs);
   }
 
-  jest.spyOn($, 'ajax').mockImplementation(function(args) {
-
+  fetch.mockImplementation(function (url, args) {
     var pair = allPairs.shift();
     if (!pair) {
       throw new Error('We are making a request that we have not anticipated.');
     }
 
     // Make sure every request is attaching cookies
-    expect(args.xhrFields).toEqual({
-      withCredentials: true
-    });
+    expect(args.credentials).toEqual('include');
 
     if (pair.request) {
-      expect(pair.request.uri).toEqual(args.url);
-      if (pair.request.data || args.data) {
-        expect(pair.request.data).toEqual(JSON.parse(args.data));
+      expect(pair.request.uri).toEqual(url);
+      if (pair.request.data || args.body) {
+        expect(pair.request.data).toEqual(JSON.parse(args.body));
       }
       if (pair.request.headers) {
         expect(pair.request.headers).toEqual(args.headers);
       }
     }
 
-    var deferred = $.Deferred();
+    var deferred = Q.defer();
```

Comment:
Since these are just tests, do we need to use Q? At some point, it'd be nice to get rid of the Q dependency in general - it was added back in the day before Promises were implemented in browsers.

Reply:
Good point! Actually, I tried using `Promise` at the beginning, then realized that linter was complaining since we don't support ES6 yet in `okta-auth-js`. So, instead of adding a lint exception, I decided to kept it consistent with existing code and opted for having complete refactor to do later in another PR.

Reply:
Gotcha, yeah it can be in another PR. As an aside - it's not uncommon to have a separate lint override file for tests vs. source (in fact, I'm guessing we have to have one anyway just to deal with test-only things). So when y'all get to it, it doesn't have to be one-shot for the whole codebase - could limit risk by first doing some/all tests, then source.

Reply:
Created some JIRAs: 
https://oktainc.atlassian.net/browse/OKTA-218652 
https://oktainc.atlassian.net/browse/OKTA-218650 
https://oktainc.atlassian.net/browse/OKTA-218649